test/cat.jl

```diff
@@ -55,13 +47,29 @@ module TestCat
     end
 
     @testset "hcat ::Vectors" begin
+        df = DataFrame()
+        DataFrames.hcat!(df, CategoricalVector{Union{Int, Missing}}(1:10))
+        @test df[1] == collect(1:10)
+        DataFrames.hcat!(df, 1:10)
+        @test df[2] == collect(1:10)
+
         df = DataFrame()
         DataFrames.hcat!(df, CategoricalVector{Union{Int, Missing}}(1:10))
         @test df[1] == CategoricalVector(1:10)
         DataFrames.hcat!(df, collect(1:10))
         @test df[2] == collect(1:10)
-    end
 
+        df = DataFrame()
+        df2 = hcat(CategoricalVector{Union{Int, Missing}}(1:10), df)
+        @test df2[1] == collect(1:10)
+        @test names(df2) == ["x1"]
```

Comment:
This will fail unless you use symbols AFAIK

Reply:
sure. fixed